subsys/bluetooth/host/id.c

```diff
@@ -410,25 +406,27 @@ static void adv_disable_rpa(struct bt_le_ext_adv *adv, void *data)
 			BT_ERR("Failed to disable advertising (err %d)", err);
 		}
 
-		adv_disabled[adv_index] = true;
-	}
+		atomic_set_bit(adv->flags, BT_ADV_RPA_UPDATE);
 
-	/* Notify the user about the RPA timeout and set the RPA validity. */
-	if (adv->cb && adv->cb->rpa_expired) {
-		rpa_invalid = adv->cb->rpa_expired(adv);
-	}
+		/* Notify the user about the RPA timeout and set the RPA validity. */
+		if (adv->cb && adv->cb->rpa_expired) {
+			rpa_invalid = adv->cb->rpa_expired(adv);
+		}
 
-	if (rpa_invalid) {
-		atomic_clear_bit(adv->flags, BT_ADV_RPA_VALID);
+		if (rpa_invalid) {
+			atomic_clear_bit(adv->flags, BT_ADV_RPA_VALID);
+		} else {
+			/* Submit the timeout in case no advertising set updates the their RPA
+			 * in the current period. This makes sure that the RPA timer is running.
+			 */
```

Comment:
```suggestion
			/* Submit the timeout in case no advertising set updates their RPA
			 * in the current period. This makes sure that the RPA timer is running.
			 */
```